File: lengths.py
# run `python lengths.py > data/lengths.csv`

import re

csv_lines = []
for version in ["test", "train", "valid"]:
	for source in ["ptb", "wikitext-103"]:
		for connective in ["because", "but"]:
			filename = "data/" + source + "/" + version + "_" + connective.upper() + ".txt"

			for line in open(filename):
				m = re.match("(.*) " + connective + " (.*)\n", line)
				if m:
					s1, s2 = m.groups()
					csv_line = ",".join([
						str(len(s1.split(" "))),
						str(len(s2.split(" "))),
						connective,
						version,
						source
					])
					csv_lines.append(csv_line)

for line in csv_lines:
	print line

File: lengths.R
df = read.csv("data/lengths.csv")
names(df) = c("len1", "len2", "connective", "version", "source")
df = df %>% mutate(file = paste(source, version)) %>%
  gather("var", "val", 1:2) %>%
  mutate(file = factor(file, levels=c(
    "ptb test",
    "ptb valid",
    "ptb train",
    "wikitext-103 test",
    "wikitext-103 valid",
    "wikitext-103 train"
  )))
df %>% filter(source =="ptb") %>%
  ggplot(., aes(x=val, colour=connective, fill=connective)) +
  geom_histogram(alpha=1/4) +
  # geom_vline(xintercept = 35) +
  facet_grid(file ~ var, scale="free")
df %>% filter(source =="wikitext-103") %>%
  ggplot(., aes(x=val, colour=connective, fill=connective)) +
  geom_histogram(alpha=1/3) +
  geom_vline(xintercept = 35) +
  facet_grid(file ~ var, scale="free") +
  xlim(0, 100)

df %>% filter(source == "ptb" & version=="train") %>%
  summarise(mean(val <= 35))

df %>% filter(source == "wikitext-103" & version=="train") %>%
  summarise(mean(val <= 35))